### PR TITLE
Added Deepweeds dataset

### PR DESCRIPTION
**Pull request by SoumyadeepJana**
Dataset Request https://github.com/tensorflow/datasets/issues/364
@Conchylicultor @cyfra @tfds-copybara @rsepassi  Kindly review :)

### CODE REVIEW

File: tensorflow_datasets/image/__init__.py [hunk 1]
```diff
@@ -26,6 +26,7 @@
 from tensorflow_datasets.image.coco import Coco2014
 from tensorflow_datasets.image.colorectal_histology import ColorectalHistology
 from tensorflow_datasets.image.colorectal_histology import ColorectalHistologyLarge
+from tensorflow_datasets.image.deep_weeds import DeepWeeds
```
Comment (us) [inline tensorflow_datasets/image/__init__.py]: You should move a bottom line.

File: tensorflow_datasets/image/deep_weeds_test.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  DATASET_CLASS = deep_weeds.DeepWeeds
+  SPLITS = {  
+      "train":4,
+  }
+ 
+
+if __name__ == "__main__":
+  tfds_test.test_main()
+
+
```
Comment (us) [inline tensorflow_datasets/image/deep_weeds_test.py]: You can delete empty lines

File: tensorflow_datasets/image/__init__.py (hunk 1 above)
Comment (ChanchalKumarMaji) [reply tensorflow_datasets/image/__init__.py]: Add in alphabetical order.

**Comment by ChanchalKumarMaji** [comment]
@SoumyadeepJana please ensure that fake_sample images are not from the original dataset.

**Comment by SoumyadeepJana** [comment]
> @SoumyadeepJana please ensure that fake_sample images are not from the original dataset.

They aren't, only the filename is same.

**Comment by us** [comment]
@cyfra @Conchylicultor being_merged.

**Pull request merged**